In [1]:
import pandas as pd


In [2]:
path = 'purchase_data.json'
purchase_data_df = pd.read_json(path)
purchase_data_df.head()

Age Gender  Item ID                                  Item Name  Price  \
0   38   Male      165                Bone Crushing Silver Skewer   3.37   
1   21   Male      119  Stormbringer, Dark Blade of Ending Misery   2.32   
2   34   Male      174                            Primitive Blade   2.46   
3   21   Male       92                               Final Critic   1.36   
4   23   Male       63                             Stormfury Mace   1.27   

             SN  
0     Aelalis34  
1        Eolo46  
2   Assastnya25  
3  Pheusrical25  
4        Aela59

In [3]:
num_players = len(purchase_data_df['SN'].unique())
num_players

573

In [4]:
unique_items = len(purchase_data_df['Item ID'].unique())
unique_items

183

In [5]:
avg_purchase_price = round(purchase_data_df['Price'].mean(), 2)
avg_purchase_price

2.93

In [6]:
num_purchases = purchase_data_df['Price'].size
num_purchases

780

In [7]:
tot_revenue = purchase_data_df['Price'].sum()
tot_revenue

2286.33

# What I need to know:

1. percentage and count of Male
2. percentage and count of Female
3. percentage and count of other / Non-Disclosed

# what I know:

1. there are more than one representation of some but not all players. the max representation is 5 times.
2. Collumn Gender contains the information needed however it has duplicates.
3. there are 573 unique players.
4. there are 780 rows in the dataframe.

# what I need to do:

1. drop the rows with duplicate players.
2. make new dataframe
3. groupby Gender
4. count gender groups
5. divide counts by total number of players to get percentages.

In [15]:
purchase_data_df.groupby('SN').count().sort_values('Gender', ascending=False)

Age  Gender  Item ID  Item Name  Price
SN                                                     
Undirrala66        5       5        5          5      5
Mindimnya67        4       4        4          4      4
Qarwen67           4       4        4          4      4
Saedue76           4       4        4          4      4
Sondastan54        4       4        4          4      4
Hailaphos89        4       4        4          4      4
Yadanun74          3       3        3          3      3
Lirtosia72         3       3        3          3      3
Seorithstilis90    3       3        3          3      3
Eusri70            3       3        3          3      3
Chanastsda67       3       3        3          3      3
Haerith37          3       3        3          3      3
Mindirra92         3       3        3          3      3
Haellysu29         3       3        3          3      3
Frichaya88         3       3        3          3      3
Frichosiala98      3       3        3          3      3
Chamistast30       3       3        3          3      3
Isursti83          3       3        3          3      3
Isurria36          3       3        3          3      3
Tillyrin30         3       3        3          3      3
Aduephos78         3       3        3          3      3
Aerithllora36      3       3        3          3      3
Sondim43           3       3        3          3      3
Aeduera68          3       3        3          3      3
Ila44              3       3        3          3      3
Chadossa56         3       3        3          3      3
Chadjask77         3       3        3          3      3
Yasriphos60        3       3        3          3      3
Lisossa63          3       3        3          3      3
Eoda93             3       3        3          3      3
...              ...     ...      ...        ...    ...
Iaralsuir44        1       1        1          1      1
Iarithdil76        1       1        1          1      1
Irith83            1       1        1          1      1
Iasur80            1       1        1          1      1
Irillo49           1       1        1          1      1
Iri67              1       1        1          1      1
Iral74             1       1        1          1      1
Inguron55          1       1        1          1      1
Inguard95          1       1        1          1      1
Ingonon91          1       1        1          1      1
Indonmol95         1       1        1          1      1
Indirrian56        1       1        1          1      1
Ina92              1       1        1          1      1
Ilrian97           1       1        1          1      1
Ilosu82            1       1        1          1      1
Ilosia37           1       1        1          1      1
Ilophos58          1       1        1          1      1
Ilirrasda54        1       1        1          1      1
Ilimya66           1       1        1          1      1
Ililsan66          1       1        1          1      1
Iliel92            1       1        1          1      1
Ilaststa70         1       1        1          1      1
Ilast79            1       1        1          1      1
Ilassa51           1       1        1          1      1
Ilariarin45        1       1        1          1      1
Ilaesudil92        1       1        1          1      1
Iduedru67          1       1        1          1      1
Idaria87           1       1        1          1      1
Idairin80          1       1        1          1      1
Zontibe81          1       1        1          1      1

[573 rows x 5 columns]

In [9]:
purchase_data_df.groupby('SN').sum().sort_values('Price', ascending=False)

Age  Item ID  Price
SN                                
Undirrala66    145      472  17.06
Saedue76       100      233  13.56
Mindimnya67    156      609  12.74
Haellysu29      63      353  12.73
Eoda93          66      284  11.58
Isursti83       72      299  11.05
Isurria36       54      390  11.01
Eusri70         90      325  10.55
Aerithllora36   87      281  10.45
Yasriphos60    120      237  10.40
Sondastan54    124      478  10.24
Haerith37       75      238  10.14
Chadjask77     108      363  10.01
Qarwen67        44      439   9.97
Sondim43        60      172   9.38
Tillyrin30      60      205   9.19
Lisistaya47     21      281   9.19
Tyisriphos58    76      282   9.18
Frichaya88      90      170   9.16
Assassasta79    34      226   9.12
Lisossan98      30      252   9.10
Yadanun74       75      373   9.09
Mindirra92      60      150   9.01
Aeliru63        44      203   8.98
Chanastsda67    45      299   8.96
Frichosiala98   75      312   8.94
Malunil62       38      201   8.89
Iskadarya95     30      200   8.79
Ilogha82        50      257   8.61
Yadaisuir65     44      115   8.56
...            ...      ...    ...
Rithe53         23       92   1.36
Pheusrical25    21       92   1.36
Isketo41        13        5   1.32
Tyaili86        17       86   1.28
Eollym91        23       86   1.28
Chamirra53      15       86   1.28
Ilaststa70      10       63   1.27
Aela59          23       63   1.27
Lassilsa63      23       63   1.27
Mindosiasya28   13      175   1.24
Raeduerin33     28      175   1.24
Heolo60         22      175   1.24
Ialidru40       19      175   1.24
Yaristi64       38      175   1.24
Ilrian97        13      175   1.24
Ryastycal90     20      122   1.21
Ilast79         20        6   1.20
Siasri67        22        6   1.20
Undirrasta74    22       41   1.16
Sundista85      22      123   1.14
Chamirrasya33   20       82   1.11
Eyircil84       23       74   1.06
Sondassa68      30       69   1.06
Lassadarsda57   24       25   1.03
Tridaira71      37       95   1.03
Rarith48        20       15   1.03
Alaesu91        18       95   1.03
Mindossa76      21       95   1.03
Lassista97      24       25   1.03
Ililsan66       18       15   1.03

[573 rows x 3 columns]